src/index.ts

```diff
@@ -0,0 +1,44 @@
+import chalk from 'chalk';
+import { chmod } from 'node:fs/promises';
+
+import { app } from '@app/app';
+import { args } from '@app/args';
+import { API_ENDPOINT } from '@app/constants';
+import { removeSocket } from '@app/utilities';
+
+import { version } from '../package.json';
+
+// we need to make sure that the socket doesn't
+// already exist, otherwise we run into issues
+// where the server can't run
+await removeSocket(args.socket);
+
+const server = {
+  fetch: app.fetch,
+  unix: args.socket,
+};
+
+Bun.serve(server);
+
+// we need to change this so that we can
+// ping the socket as a non-privileged user
```

Comment:
Maybe this comment could be a bit extended (or shortened)? I wonder what the plan here is about non-privileged users.

AIUI in osbuild-composer we had 0660 as well with root:weldr for the socket. That seems sensible to me. If we switch to a model where non-privileged users (without being in the weldr group) can talk to the socket we need to be careful on the server side because our code there runs as root and giving anyone unrestricted access to image-building is a departure from what we traditionally did (and with the ability to inject custom packages into the build its also a potential security risk). So its a much bigger step than what the comment implies (at least what it implies to me :)

Reply:
Sounds fine to me, I assume when you say: "remove this change completely" you mean the comment, yes? because I think thats all that is needed.

About the frustration/testing - maybe add a --socket-path cmdline and then you can just point it to your own local dir (or your own tempdir). Would that work to help with the frustration?

Reply:
Oh ya sorry of course, we changed the implementation but I forgot to change the comment. I'll remove it :)

Ya I think using a different socket path will solve that and we accept it as a CLI arg or environment variable already.